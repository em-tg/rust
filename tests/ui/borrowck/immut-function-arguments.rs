//@ check-pass
fn f(y: Box<isize>) {
    *y = 5; //~ WARNING cannot assign
}

fn g() {
    let _frob = |q: Box<isize>| { *q = 2; }; //~ WARNING cannot assign
}

fn main() {}
