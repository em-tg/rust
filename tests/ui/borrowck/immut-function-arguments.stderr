warning[E0594]: cannot assign to `*y`, as `y` is not declared as mutable
  --> $DIR/immut-function-arguments.rs:3:5
   |
LL |     *y = 5;
   |     ^^^^^^ cannot assign
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider changing this to be mutable
   |
LL | fn f(mut y: Box<isize>) {
   |      +++

warning[E0594]: cannot assign to `*q`, as `q` is not declared as mutable
  --> $DIR/immut-function-arguments.rs:7:35
   |
LL |     let _frob = |q: Box<isize>| { *q = 2; };
   |                                   ^^^^^^ cannot assign
   |
help: consider changing this to be mutable
   |
LL |     let _frob = |mut q: Box<isize>| { *q = 2; };
   |                  +++

warning: 2 warnings emitted

For more information about this error, try `rustc --explain E0594`.
